lib/Tomcat/Utils.pm

```diff
@@ -68,20 +71,28 @@ EOF
 }
 
 
-# Access the tomcat web application manager
+# Check Servlet, JSP and Websocket, the example files can be accessed, login with admin via tomcat manager
 sub tomcat_manager_test() {
-    my ($self) = shift;
-
-    $self->firefox_open_url('localhost:8080/manager');
-    assert_screen('tomcat-manager-authentication', 180);
-    type_string('admin');
-    send_key('tab');
-    type_string('admin');
-    assert_and_click('tomcat-OK-autentication');
-    wait_still_screen(2);
-    assert_and_click('tomcat-click-save-login');
-    assert_screen('tomcat-web-application-manager', TIMEOUT);
+    select_serial_terminal;
+
+    # sometimes we have sporadi issue with connection to localhost for unknown reason
+    # just a short check with ping
+    assert_script_run('ping -c 4 localhost');
```

Comment:
Hello Zaoliang,

May you please have a try to restart the network service if connection to localhost is lost? and then quit the test if the network can't still work.

Reply:
this is dimstar already mentioned before, restarting network.service is hammer :) so I can accept this without to start it, but at least we know where is problem by checking ping localhost. So we can still fix easy and fast the kind of sporadic issue in future. At moment I cannot determine why connect to localhost doesn't work sometimes :)
I assume at moment it is working... we can leave this now.